src/Microsoft.AspNetCore.HttpsPolicy/HttpsRedirectionMiddleware.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Hosting.Server.Features;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.Extensions.Options;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.HttpsPolicy
+{
+    public class HttpsRedirectionMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private int? _httpsPort;
+        private readonly int _statusCode;
+
+        private readonly IServerAddressesFeature _serverAddressesFeature;
+        private bool _evaluatedServerAddressesFeature;
+
+        /// <summary>
+        /// Initializes the HttpsRedirectionMiddleware
+        /// </summary>
+        /// <param name="next"></param>
+        /// <param name="serverAddressesFeature">The</param>
+        /// <param name="options"></param>
+        public HttpsRedirectionMiddleware(RequestDelegate next, IServerAddressesFeature serverAddressesFeature, IOptions<HttpsRedirectionOptions> options)
+        {
+            if (next == null)
+            {
+                throw new ArgumentNullException(nameof(next));
+            }
+
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+            _next = next;
+
+            var httpsRedirectionOptions = options.Value;
+            _httpsPort = httpsRedirectionOptions.HttpsPort;
+            _statusCode = httpsRedirectionOptions.RedirectStatusCode;
+            _serverAddressesFeature = serverAddressesFeature;
+        }
+
+        /// <summary>
+        /// Invokes the HttpsRedirectionMiddleware
+        /// </summary>
+        /// <param name="context"></param>
+        /// <returns></returns>
+        public Task Invoke(HttpContext context)
+        {
+            if (!context.Request.IsHttps)
+            {
+                if (!_evaluatedServerAddressesFeature && _httpsPort == null)
+                {
+                    CheckAddressesFeatureForHttpsPorts();
+                }
+                _evaluatedServerAddressesFeature = true;
+
+                var host = context.Request.Host;
+                if (_httpsPort.HasValue && _httpsPort.Value > 0)
```

Comment:
It should always have a value by the time you get here.
how would it be set to 0? Shouldn't that be validated earlier?

Reply:
I copied this from UrlRewrite, probably was wrong there.

Reply:
Also, httpsPort not having a value implies 443 (empty)?